internal/loader/configuration_client_manager.go

```diff
@@ -461,3 +475,70 @@ func Jitter(duration time.Duration) time.Duration {
 	// Apply the random jitter to the original duration
 	return duration + time.Duration(randomJitter)
 }
+
+func newClientAssertionCredential(ctx context.Context, serviceAccountName string, serviceAccountNamespace string) (azcore.TokenCredential, error) {
+	cfg, err := ctrlcfg.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := client.New(cfg, client.Options{})
+	if err != nil {
+		return nil, err
+	}
+
+	serviceAccountObj := &corev1.ServiceAccount{}
+	err = client.Get(ctx, types.NamespacedName{Namespace: serviceAccountNamespace, Name: serviceAccountName}, serviceAccountObj)
+	if err != nil {
+		return nil, err
+	}
+
+	tenantId, ok := os.LookupEnv(strings.ToUpper(AzureTenantId))
+	if !ok {
+		return nil, fmt.Errorf("no tenant ID specified. Check pod configuration or set TenantID in the options")
+	}
+
+	if _, ok := serviceAccountObj.Annotations[AnnotationClientID]; !ok {
+		return nil, fmt.Errorf("service account %s/%s does not have annotation %s", serviceAccountNamespace, serviceAccountName, AnnotationClientID)
+	}
+
+	if _, ok := serviceAccountObj.Annotations[AnnotationTenantID]; ok {
+		tenantId = serviceAccountObj.Annotations[AnnotationTenantID]
+	}
+
+	getAssertionFunc := newGetAssertionFunc(serviceAccountNamespace, serviceAccountName)
+
+	clientAssertionCredential, err := azidentity.NewClientAssertionCredential(tenantId, serviceAccountObj.Annotations[AnnotationClientID], getAssertionFunc, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return clientAssertionCredential, nil
+}
+
+func newGetAssertionFunc(serviceAccountNamespace string, serviceAccountName string) func(ctx context.Context) (string, error) {
+	audiences := []string{AzureDefaultAudience}
```

Comment:
the name is a little misleading. it's not default audience of Azure. I believe https://management.azure.com is the actual default for control plane. `ApiTokenExchangeAudience`?

Reply:
Make sense, updated.